src/machine/heap.rs

```diff
@@ -1044,31 +1041,23 @@ impl Heap {
 
     /// Returns the number of bytes needed to store `src` as a `PStr`.
     pub(crate) fn compute_pstr_size(src: &str) -> usize {
+        if src.is_empty() {
+            // As of now, null strings aren't allocated.
```

Comment:
The string `""` is the atom `[]`, and that atom should be used to represent the string.

<pre>
<b>?- "" == [].</b>
   true.
</pre>

Reply:
I've amended the comment to now say:

> Empty strings are represented by the '[]' atom.
> As such, no PStr is needed or allocated: